meteor/sofie-server/model/entities/test/rundown.spec.ts

```diff
@@ -0,0 +1,1462 @@
+import { Segment, SegmentInterface } from '../segment'
+import { Rundown, RundownInterface } from '../rundown'
+import { Part, PartInterface } from '../part'
+import { Piece, PieceInterface } from '../piece'
+import { PieceLifespan } from '../../enums/piece-lifespan'
+import { EntityMockFactory } from './entity-mock-factory'
+import { capture, instance, verify, when } from 'ts-mockito'
+
+describe('Rundown', () => {
+	describe('instantiate already active Rundown', () => {
+		describe('"alreadyActiveProperties" is provided', () => {
+			describe('active status is provided as false', () => {
+				it('throws error', () => {
```

Comment:
Tests ensuring that the correct exceptions are thrown, should propably make use of expect in some way, possibly as i have done/shown.
This way you can ensure that it is the correct error and/or message being thrown.

```typescript
{
   ...
   expect.assertions(1)
   try {
      new Rundown(rundownInterface)
   } catch (error) {
       expect(error).toBeInstanceOf(MisconfigurationException)
   }
}
```

Reply:
`expect.assertions` is not needed.

The way `expect` works is by throwing an Error if the condition isn't met. If the condition is met, it does nothing. Just like the implementation here.

Reply:
> Would also stop GitHub actions from complaining out missing assertions.

That's lint, not GitHub and could fixed by a change to the linting rule if wanted

Reply:
From what you are saying, am i then right in that you want me to change my setups, to not use the `expect.assertions(...)`?

Reply:
In any case, adding a check to ensure the error thrown is of the correct type, would still be good to do, in my mind.

Reply:
> From what you are saying, am i then right in that you want me to change my setups, to not use the expect.assertions(...)?

Only if you want to. Personally I prefer not to use the `expect.assertions(...)` but it's a different way of achieving the same thing and both ways are acceptable.

The more descript error message would be nice though ;)

Reply:
> instead throw an descriptive error, should it not throw the expected exception.

With the possibility of misunderstanding what you meant: 
The test that test that an error is thrown expects an exception to happen. If an exception happens, then our test should pass. In order for the test to pass, it should throw no errors.

So if you rethrow the expected exception then the test would be marked as failed.

We want the test to fail when no exception is thrown, so there would be no exception to rethrow when the test should fail.

Reply:
I had changed it so that if it succeded to get past the try/catch, without hitting the catch, then it would throw an error with a message explaining what happend, as we'd expect the catch to be hit.

This can however be ignored thanks to a god suggestion by Krzy, that eliminated the try/catch, in favor of using `await expect(...).reject.toThrow(...)`

Fell free to resolve, ones your are happy.